estate_sale/models/sale_order.py

```diff
@@ -26,3 +26,38 @@ def send_internal_email(self):
     def _compute_big_order(self):
         for record in self:
             record.is_big_order = record.amount_total > 1000
+
+    def action_duplicate_lines_to_new_order(self):
+        for order in self:
+            # create a new order
+            new_order = self.env["sale.order"].create(
+                {
+                    "partner_id": order.partner_id.id,
+                    "user_id": order.user_id.id,
+                    "origin": f"Copied from {order.name}",
+                }
+            )
+
+            # create order lines (without discounts)
+            for line in order.order_line:
+                self.env["sale.order.line"].create(
+                    {
+                        "order_id": new_order.id,
+                        "product_id": line.product_id.id,
+                        "product_uom_qty": line.product_uom_qty,
+                        "product_uom": line.product_uom.id,
+                        "price_unit": line.price_unit,
+                        "discount": 0.0,
+                        "name": line.name,
+                    }
+                )
```

Comment:
For performance reasons, it's better to create records in a bunch if possible. First, we need to gather list of values dicts, then create. This way we perform only one write transaction in the DB, instead of perform write transaction for each element in the loop. 
https://www.odoo.com/documentation/18.0/developer/reference/backend/performance.html#batch-operations
```suggestion
line_vals = []
for line in order.order_line:
    line_vals.append(
        {
            "order_id": new_order.id,
            "product_id": line.product_id.id,
            "product_uom_qty": line.product_uom_qty,
            "product_uom": line.product_uom.id,
            "price_unit": line.price_unit,
            "discount": 0.0,
            "name": line.name,
        }
    )
    
self.env["sale.order.line"].create(line_vals)
```